jhack/conf/conf.py

```diff
@@ -85,4 +92,81 @@ def print_current_config():
     CONFIG.pprint()
 
 
+class Denied:
+    def __bool__(self):
+        return False
+
+
+class _Reason(str, Enum):
+    confirmed = "confirmed"
+    devmode = "devmode"
+
+
+class Allowed:
+    def __init__(self, reason: _Reason):
+        self.reason = reason
+
+    def __bool__(self):
+        return True
+
+
+def check_destructive_commands_allowed(
+    msg: str, dry_run_cmd: str = "", _check_only=False
+) -> Union[Denied, Allowed]:
+    if os.getenv("JHACK_PROFILE") == "devmode":
+        logger.debug(f"operation {msg} allowed by devmode profile.")
+        return Allowed(_Reason.devmode)
+
+    if not CONFIG.get(
+        "general", "enable_destructive_commands_NO_PRODUCTION_zero_guarantees"
+    ):
+        preamble = (
+            "in order to run this command without confirmation prompt, you must enable destructive mode. "
+            "this mode is intended for development environments and should be disabled "
+            "in production! This is *for your own good*. "
+        )
+        closure = (
+            "Or, if you want to allow destructive mode just this once, set the "
+            "`JHACK_PROFILE=devmode` envvar."
+        )
```

Comment:
Perhaps the command that was run with the `JHACK_PROFILE=devmode` could be printed so that someone can copy and paste to run it

Reply:
as in, in addition to also running it?
if you want to see the commands to run them manually, that's what dry_run would be for.

Reply:
oh ok I get it, will see if I can get it done